library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -627,6 +627,23 @@ def CanonicalizeIP(ifcfg)
       ifcfg
     end
 
+    # Filters out INTERFACETYPE option from ifcfg config when it is not needed.
+    #
+    # INTERFACETYPE has big impact on wicked even yast behavior. It was overused
+    # by yast in the past. According wicked team it makes sense to use it only
+    # in two cases 1) lo device (when it's name is changed - very strongly discouraged)
+    # 2) dummy device
+    #
+    # This function silently modifies user's config files. However, it should make sense
+    # because:
+    # - INTERFACETYPE is usually not needed
+    # - other functions in this module modifies the config as well (see Canonicalize* functions)
+    # - using INTERFACETYPE is reported as a warning by wicked (it asks for reporting a bug)
+    # - it is often ignored by wicked
```

Comment:
\# @return [Hash] filtered devmap
`delete_if` returns the new hash and also modifies the argument. To reduce surprises, please start the body with
```rb
devmap = deep_copy(devmap)
```